src/main/java/com/iexec/sms/admin/AdminController.java

```diff
@@ -175,14 +201,19 @@ private ResponseEntity<Void> performOperation(String storageID, String fileName,
                     operationSuccessful = adminService.createDatabaseBackupFile(adminStorageLocation + BACKUP_STORAGE_LOCATION, BACKUP_FILENAME);
                     break;
                 case RESTORE:
-                    operationSuccessful = adminService.restoreDatabaseFromBackupFile(storagePath, fileName);
+                    operationSuccessful = adminService.restoreDatabaseFromBackupFile(sourceStoragePath, sourceFileName);
                     break;
                 case DELETE:
-                    operationSuccessful = adminService.deleteBackupFileFromStorage(storagePath, fileName);
+                    operationSuccessful = adminService.deleteBackupFileFromStorage(sourceStoragePath, sourceFileName);
                     break;
                 case REPLICATE:
                     operationSuccessful = adminService.replicateDatabaseBackupFile(
-                            adminStorageLocation + BACKUP_STORAGE_LOCATION, BACKUP_FILENAME, storagePath, fileName);
+                            adminStorageLocation + BACKUP_STORAGE_LOCATION, BACKUP_FILENAME, sourceStoragePath, sourceFileName);
```

Comment:
```suggestion
                            adminStorageLocation + BACKUP_STORAGE_LOCATION, BACKUP_FILENAME, destinationStoragePath, destinationFileName);
```

Reply:
global remark, I see `REPLICATE` as a particular `COPY` action. It should be possible to share some code between the two actions.